data_storage/src/data_storage_service.py

```diff
@@ -0,0 +1,126 @@
+from kafka import KafkaConsumer, TopicPartition
+from elastic_handler import ElasticsearchHandler
+import json
+import time
+
+
+class Message:
+    def __init__(self, key, value, partition, offset):
+        self.key = key
+        self.value = value
+        self.partition = partition
+        self.offset = offset
+
+    def __repr__(self) -> str:
+        return f"Message(key={self.key}, value={self.value}, partition={self.partition}, offset={self.offset})"
+    
+class KafkaHandler:
+    """Handles Kafka consumer setup and operations."""
+    
+    def __init__(self, topic, bootstrap_servers, group_id, auto_offset_reset='earliest'):
+        self.topic = topic
+        self.bootstrap_servers = bootstrap_servers
+        self.group_id = group_id
+        self.auto_offset_reset = auto_offset_reset
+        self.consumer = None
+        
+
+    def setup_consumer(self, value_deserializer, key_deserializer):
+        """Initializes the Kafka consumer."""
+        self.consumer = KafkaConsumer(
+            self.topic,
+            bootstrap_servers=self.bootstrap_servers,
+            auto_offset_reset=self.auto_offset_reset,
+            enable_auto_commit=False,
+            group_id=self.group_id,
+            value_deserializer=value_deserializer,
+            key_deserializer=key_deserializer
+        )
+        return self.consumer
+
+    @staticmethod
+    def wait_for_kafka(consumer, retries=2, delay=1):
+        """Waits for Kafka to be ready, retrying if necessary."""
+        for _ in range(retries):
+            try:
+                consumer.poll(timeout_ms=1000)
+                return True
+            except Exception as e:
+                print(f"Waiting for Kafka... Error: {e}")
+                time.sleep(delay)
+        return False
+
+    def consume_messages(self, max_messages=None):
+        """Consumes messages from the Kafka topic."""
+        if not self.consumer:
+            raise RuntimeError("Consumer has not been set up. Call setup_consumer first.")
+        
+        if self.wait_for_kafka(self.consumer):
+            print("Connected to Kafka.")
+            self.read_from_begining()
```

Comment:
Chyba nie powinnismy zawsze czytać od początku topicu, co nie?

Reply:
Tak, to tylko dla testow zeby przy kazdym odpaleniu nie pisac wiadomosci od nowa